### PR TITLE
Remove build-essential

### PR DESCRIPTION
**Pull request by adamdecaf**
This drops the image down to `216.4 MB`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -7,13 +7,13 @@ ADD collectd.conf /etc/collectd/
 ADD entrypoint.sh /entrypoint.sh
 
 RUN echo 'ef97838e249814ad96a2edc0410326efbb4f3d08  collectd-5.5.0.tar.bz2' > /tmp/collectd.sig && \
-    apt-get update && \
-    apt-get install -y build-essential wget tar lbzip2 && \
+    apt-get update && apt-get install -y build-essential wget tar lbzip2 && \
```
Comment (BusyByte) [inline Dockerfile:10]: @adamdecaf so you are installing build-essentials and then removing it after you configure collectd?

**Comment by BusyByte** [comment]
What was the size before?

**Comment by BusyByte** [comment]
👍 regardless of my questions

**Comment by sherzberg** [comment]
👍

**Comment by adamdecaf** [comment]
On the original fork we used it was close to 900MB. Overtime we've shrunk the container down. There's a previous changeset that dropped it to [~350MB](https://github.com/Banno/docker-collectd/pull/1).

File: Dockerfile (hunk 1 above)
Comment (adamdecaf) [reply Dockerfile:10]: Yep. The way docker containers create the underlying file system is to basically store the diff between Dockerfile commands. This means if you have a lot of commands apart from each other then they are stored as differnet layers.

```
RUN apt-get update
RUN apt-get upgrade -y 
```

The above is two differnet layers, each stored with the diff of the changed fs. The goal to keep containers manageable is to have as small of an underlying fs diff as possible. This means doing lots of things in one step (and then cleaning up after). 

http://developers.redhat.com/blog/2016/03/09/more-about-docker-images-size/

**Pull request merged**